### DISCUSSION TITLE
[Dev Tooling] Verification: the provided chain with id 531050104 is not supported by default

### DISCUSSION THREAD

**Discussion by lutr0** [zkSync-Community-Hub/zksync-developers#783] [Dev Tooling] [answered]
### Environment

Testnet

### Select the Dev Tool you are using

Hardhat Plugins

### Provide the version of the tool (if applicable)

"@matterlabs/hardhat-zksync-upgradable": "^1.8.1",

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

We are getting error when trying to verify a UUPS proxy deployed on Sophon testnet (https://explorer.testnet.sophon.xyz/address/0x6a2ab154d7c5Ba9fdea6d8A0C79818A4463a63f9)
```
npx hardhat --network sophonTestnet verify 0x6a2ab154d7c5Ba9fdea6d8A0C79818A4463a63f9


Verifying implementation: 0xB4f1b61225Af35Ab5c02a3Aa10864Ec650DeAD00
ZkSyncVerifyPluginError: The provided chain with id 531050104 is not supported by default!
Verifying proxy: 0x854ee52aC916E2F18AdC4372391889d10ab4eBc5
No logs found for event topic AdminChanged(address,address) at address 0x854ee52aC916E2F18AdC4372391889d10ab4eBc5
One of possible reasons can be that you are trying to verify a UUPS contract
The provided chain with id 531050104 is not supported by default!
```
I never encountered this error before. Attaching the code below.
The contracts are successfully deployed

### Repo Link (Optional)

https://github.com/Stork-Oracle/stork-external/pull/54/files

**Comment by alhenaai** [comment]
It appears that the error you're encountering is related to the chain ID `531050104` not being supported by default in the verification plugin you're using. This issue typically arises when the verification tool does not recognize or support the network specified in your Hardhat configuration.

Here are a few steps you can take to troubleshoot and potentially resolve this issue:

1. **Check Network Configuration**: Ensure that your Hardhat configuration (`hardhat.config.js` or `hardhat.config.ts`) includes the correct settings for the `sophonTestnet`. Verify that the network URL and chain ID are correctly specified.

2. **Update the Plugin**: Since you are using the `@matterlabs/hardhat-zksync-upgradable` plugin, make sure it is up to date. Sometimes, newer versions of plugins add support for additional networks or fix bugs related to network compatibility.

3. **Custom Network Support**: If `sophonTestnet` is a custom network or not widely supported, you might need to manually add support for this network in your verification tool or script. This could involve configuring the plugin to recognize the custom chain ID.

4. **Consult Plugin Documentation**: Review the documentation for the `@matterlabs/hardhat-zksync-upgradable` plugin to see if there are specific instructions or configurations required for verifying contracts on less common networks.

5. **Reach Out for Support**: Since this seems to be a more intricate issue related to network support in a development tool, consider creating a discussion on the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) to seek advice from other developers or the tool maintainers.

By following these steps, you should be able to identify and possibly resolve the issue with verifying the UUPS proxy on the `sophonTestnet`.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by lutr0** [comment, marked as answer]
Got the fix:
disable the etherscan object temporarly with
```
etherscan: {
    enabled: false,
    apiKey: {
      ........
    }
}
```
